optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/uni/BavetGroupBridgeUniNode.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet.uni;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.optaplanner.core.api.score.stream.uni.UniConstraintCollector;
+import org.optaplanner.core.impl.score.stream.bavet.BavetConstraintSession;
+import org.optaplanner.core.impl.score.stream.bavet.bi.BavetGroupBiNode;
+import org.optaplanner.core.impl.score.stream.bavet.bi.BavetGroupBiTuple;
+import org.optaplanner.core.impl.score.stream.bavet.common.BavetTupleState;
+
+public final class BavetGroupBridgeUniNode<A, GroupKey_, ResultContainer_, Result_> extends BavetAbstractUniNode<A> {
+
+    private final BavetAbstractUniNode<A> parentNode;
```

Comment:
Unused

Reply:
All nodes know their parent currently (just like the tuples know their parents). That's by design and for consistently currently. Once this work fleshes out and this turns out to be unneeded - we can discuss removing them.